### PR TITLE
Gives detectives an ID that looks like an assistant ID when flipped. Also makes their camera silent.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
This PR adds an ID subtype which the detective outfit uses. It that can be flipped to go plainclothes as an assistant. Access retained. Spare clothes not included. 

I've also made the detective camera silent (a sound cue is still played to the user, others won't hear it) and with the flash disabled.

## Why It's Good For The Game
I want to make it a little easier for detectives to go plainclothes. Between clothings and inventory management, it's already quite tricky as a gimmick and without doubt a double-edged sword that can lead to goofy-ass situations where officers actually confuse you for an assistant. As for the camera, it's just a stealth buff.

## Changelog

:cl:
add: Gave the detective an ID that can flipped to look like an assistant ID.
balance: The detective camera is now silent and doesn't flash.
/:cl:

### CODE REVIEW

**Comment by MrMelbert** [comment]
taking hop job content

File: code/modules/photography/camera/other.dm [hunk 1]
```diff
@@ -25,7 +25,13 @@
...
+	name = "detective's camera"
+	desc = "A silent polaroid camera with extra capacity for crime investigations."
+	flash_enabled = FALSE
+	silent = TRUE
 	pictures_max = 30
 	pictures_left = 30
+
+/obj/item/camera/detective/after_picture(mob/user, datum/picture/picture)
+	. = ..()
+	user.playsound_local(src, pick('sound/items/polaroid1.ogg', 'sound/items/polaroid2.ogg'), 35, TRUE)
```
Comment (MrMelbert) [inline code/modules/photography/camera/other.dm:37]: You need to pass a turf to `playsound_local` or it won't sound as you would expect
```suggestion
	user.playsound_local(get_turf(src), pick('sound/items/polaroid1.ogg', 'sound/items/polaroid2.ogg'), 35, TRUE)
```

**Comment by TheVekter** [comment]
I get where Melbert's coming from, but I also think there's a concern here about how players are going to react to a plainclothes detective. At least a few rules on our server rely on security being identified as security.

I'm not saying this shouldn't be merged (I actually love the idea), but if it is, we might have to have a conversation about how we want to handle these sorts of situations.

File: code/game/objects/items/cards_ids.dm [hunk 2]
```diff
@@ -1414,6 +1419,44 @@
...
+	else
+		. += span_smallnotice("flip it to switch to the plainclothes identity.")
+
+/obj/item/card/id/advanced/plainclothes/attack_self(mob/user)
+	var/popup_input = tgui_input_list(user, "Choose Action", "Two-Sides ID", list("Show", "Flip"))
+	if(!popup_input || !after_input_check(user))
+		return TRUE
+	if(popup_input == "Show")
+		return ..()
+	balloon_alert(user, UNLINT("ID flipped"))
```
Comment (ZephyrTFA) [inline code/game/objects/items/cards_ids.dm:1445]:
```suggestion
	balloon_alert(user, "flipped")
```

**Comment by SomeoneYouProbablyKnow** [comment]
> it's already quite tricky as a gimmick and without doubt a double-edged sword that can lead to goofy-ass situations where officers actually confuse you for an assistant

So you're reducing the amount of fun moments for the gimmick? Your disguise being so good you get arrest is funny as hell. The double sided ID thing is kinda fire tho, could for sure slap it on like an agent ID.

And also i personally hate that gimmick because 9/10 times it's just a seccie abusing the fact that valids don't expect an assistant to pull out a gun and a baton. Not like there's much evidence collecting as sad as that is.

**Comment by LemonInTheDark** [comment]
> I'm not saying this shouldn't be merged (I actually love the idea), but if it is, we might have to have a conversation about how we want to handle these sorts of situations.

Should elaborate on your worries. Would this be a policy forum kinda deal? Ideally we want to provide detectives the space to do this because it fits the "fantasy" so well.

**Comment by Ghommie** [comment]
> taking hop job content

Yeah, I get you. That's a reason I don't want this to be anyhow customizable (outside of var-editing). It's pretty basic overall, and it doesn't even hide the real name, so if they need anything less blatant, they'd still have to visit the HoP.

> At least a few rules on our server rely on security being identified as security.

Never heard of that rule until now, perhaps I'm a tad out-of-date. Yet, I think it's been said several time that the detective isn't exactly a secoff, even if they oft end up acting as one. Whatever that fictional or not policy is, it kinda furthers the idea that detectives are just another flavor of secoffs.

> 
> I'm not saying this shouldn't be merged (I actually love the idea), but if it is, we might have to have a conversation about how we want to handle these sorts of situations.

> So you're reducing the amount of fun moments for the gimmick? Your disguise being so good you get arrest is funny as hell. The double sided ID thing is kinda fire tho, could for sure slap it on like an agent ID.

No, I didn't mean anything like that. What I meant is that it isn't just a plain detective buff, because this sort of accidents can effectively happen.

> And also i personally hate that gimmick because 9/10 times it's just a seccie abusing the fact that valids don't expect an assistant to pull out a gun and a baton. Not like there's much evidence collecting as sad as that is.

A reason why the ID is quite unique, and cannot be found anywhere else outside of the occasional mail: I don't want other seccie to have it for the valid-hunting schemes.

**Comment by zxaber** [comment]
I've seen (and dealt with administratively) a few officers that like to go plainclothes. The way I always handle it is that if an officer doesn't have a sec ID showing *and* a sec jumpsuit on, they lose all security meta-protections. They cannot assume ill intent for those resisting arrest (or others assisting such) if undercover, and have to accept the risks of a suspect fighting back.

**Comment by YakumoChen** [comment]
Dets are implanted so it should be easy for security to tell apart a plainclothes detective to be fair

**Comment by SomeoneYouProbablyKnow** [comment]
Yes. Which is exactly what ends up happening, a an assistant pulls out sec gear and before you know it very sec in the area is on their side because they have the funny implant. Which is why 9/10 times it's a trash gimmick and this codifies it.

**Comment by DrDiasyl** [comment]
> Yes. Which is exactly what ends up happening, a an assistant pulls out sec gear and before you know it very sec in the area is on their side because they have the funny implant. Which is why 9/10 times it's a trash gimmick and this codifies it.

Probably a good reason to not trust every random Assistant on this station as a antagonist. It is a _paranoia-laden_ game and such elements have a part in this game.

**Comment by ExcessiveUseOfCobblestone** [comment]
> Yes. Which is exactly what ends up happening, a an assistant pulls out sec gear and before you know it very sec in the area is on their side because they have the funny implant. Which is why 9/10 times it's a trash gimmick and this codifies it.

Is the issue that you lose to the assistant because he isnt an assistant, or is the issue that the assistant who isnt an assistant is causing trouble and sec is on their side because of their implant?

As for the admin side, there's no rule that prohibits them from doing it this way now with a wallet and a duplicate ID from the HOP. Shouldn't cause any new problems because this is just a feature that cuts out the middleman. I'm no longer an admin but it seems like the answer even if this was new ground is that sec obviously doesnt get the protections when theyre not appearing as sec but still retain their responsibilities, so someone who is playing pro-sec detective either needs to prove they are sec to the people around them if theyre ballsy enough to do something like arrest a person, OR they need to be held under the scrutiny of sec if theyre redtiding under the guise of sec.

**Comment by TheVekter** [comment]
>... so someone who is playing pro-sec detective either needs to prove they are sec to the people around them if theyre ballsy enough to do something like arrest a person, OR they need to be held under the scrutiny of sec if theyre redtiding under the guise of sec.

Compelling reason to finally add a badge for Detectives. I might very well do that if I can get a good sprite.

**Comment by SomeoneYouProbablyKnow** [comment]
> Is the issue that you lose to the assistant because he isnt an assistant, or is the issue that the assistant who isnt an assistant is causing trouble and sec is on their side because of their implant?

Mostly my issue is that due to how mindshields work even if we have rules on escalation against sec and unlawful arrests what you have in reality is sec ganging up to defend most mindshielded individuals (which is a reasonable move given it's one of the few constants in a game with limited information, which is why impersonating sec as an antag is insanely powerful). So what ends up happening is an assistant starts trying to arrest you and if you fight back you'll all the kings horses and all the mings man on your ass.
The gimmick isn't really fun because most of the time an assistant batons you out of nowhere. There's no leadup and no roleplay.

> As for the admin side, there's no rule that prohibits them from doing it this way now with a wallet and a duplicate ID from the HOP. Shouldn't cause any new problems because this is just a feature that cuts out the middleman.

Sure. Objectively correct statement. But giving sec butcher cleavers would also fall under cutting out the middle man in this analogy. Roundstart gear matters because it sets the social expectations for what you do as this job. "It's actually already easy to do" is not gonna convince me when i personally hate the thing

> OR they need to be held under the scrutiny of sec if theyre redtiding under the guise of sec.

That would be fun but sec's behavior is near indistinguishable from real police and they'll thin red line each other even if they disagree because again they are the only 100% certain allies.

Frankly my beef is overexagerrated and unimportant but i do love whining about it :nail_care:

**Pull request merged**